File: WoofIOS/SwiftPackages/Presentation/Profile/Sources/PresentationProfile/UI/ProfilePagerAdapter.swift
import SwiftUI

public struct ProfilePagerAdapter: View {
    @StateObject private var viewModel: ProfilePagerViewModel

    public init(dogId: Int) {
        _viewModel = .init(arg1: dogId)
    }

    public var body: some View {
        ProfilePagerScreen(state: viewModel.state) { page in
            ProfileViewAdapter(page: page)
        }
        .onAppear {
            viewModel.onViewAppear()
        }
    }
}

File: WoofAndroid/presentation/profile/src/main/kotlin/com/perrystreet/woof/presentation/profile/ui/ProfilePagerAdapter.kt
package com.perrystreet.woof.presentation.profile.ui

import androidx.compose.runtime.Composable
import androidx.compose.runtime.getValue
import com.perrystreet.woof.presentation.profile.viewmodel.ProfilePagerViewModel
import org.koin.androidx.compose.koinViewModel
import org.koin.core.parameter.parametersOf

@Composable
fun ProfilePagerAdapter(
    dogId: Long,
    viewModel: ProfilePagerViewModel = koinViewModel(key = "profile.pager:$dogId") { parametersOf(dogId) },
) {
    val state by viewModel.state.subscribeAsState()

    ProfilePagerScreen(state = state) { page ->
        ProfileViewAdapter(page = page)
    }
}
